pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -32,31 +32,27 @@ import (
 
 type controlPlaneInstanceReconciler struct {
 	common.ControllerResources
-	Instance       *v1.ServiceMeshControlPlane
-	Status         *v1.ControlPlaneStatus
-	ownerRefs      []metav1.OwnerReference
-	meshGeneration string
-	renderings     map[string][]manifest.Manifest
-	lastComponent  string
-	cniConfig      cni.Config
+	Instance          *v1.ServiceMeshControlPlane
+	Status            *v1.ControlPlaneStatus
+	ownerRefs         []metav1.OwnerReference
+	meshGeneration    string
+	renderings        map[string][]manifest.Manifest
+	waitForComponents sets.String
+	cniConfig         cni.Config
 }
 
 // ensure controlPlaneInstanceReconciler implements ControlPlaneInstanceReconciler
 var _ ControlPlaneInstanceReconciler = &controlPlaneInstanceReconciler{}
 
 // these components have to be installed in the specified order
-var orderedCharts = []string{
-	"istio", // core istio resources
-	"istio/charts/security",
-	"istio/charts/prometheus",
-	"istio/charts/tracing",
-	"istio/charts/galley",
-	"istio/charts/mixer",
-	"istio/charts/pilot",
-	"istio/charts/gateways",
-	"istio/charts/sidecarInjectorWebhook",
-	"istio/charts/grafana",
-	"istio/charts/kiali",
+var orderedCharts = [][]string{
+	{"istio"}, // core istio resources
+	{"istio/charts/security", "istio/charts/galley"}, // both are deployed at the same time because each is a dependency of the other
```

Comment:
I agree with @rcernich that this one could be problematic. If the webhookconfig from `galley` is already present when the istio.io resources from `security`, but the `galley` pod isn't ready yet, we're going to see errors

Reply:
I just realized that from 1.6 onwards upstream does not deploy istio.io configuration on installation anymore, so this problem will likely disappear. I suggest we postpone the merge of security and galley until then

Reply:
We'll still need this for backward compatibility.